src/pages/s/[classID]/teachers.tsx

```diff
@@ -0,0 +1,362 @@
+            id: 3,
+            prefix: "master",
+            name: {
+              "en-US": { firstName: "Thanakorn", lastName: "Arjanawut" },
+              th: { firstName: "ธนกร", lastName: "อรรจนาวัฒน์" },
+            },
+            profile: "/images/dummybase/thanakorn.webp",
+            role: "teacher",
+            classAdvisorAt: {
+              id: 1,
+              name: {
+                "en-US": "M.405",
+                th: "ม.405",
+              },
+            },
+            subjectsInCharge: [
+              {
+                id: 3,
+                code: {
+                  "en-US": "SCI6969",
+                  th: "ว6969",
+                },
+                name: {
+                  "en-US": {
+                    name: "Core Science",
+                  },
+                  th: { name: "วิทยาศาสตร์พื้นๆ" },
+                },
+                subjectSubgroup: {
+                  name: { "en-US": "Science", th: "วิทยาศาสตร์" },
+                  subjectGroup: {
+                    name: { "en-US": "Science", th: "วิทยาศาสตร์" },
+                  },
+                },
+              },
+              {
+                id: 4,
+                code: {
+                  "en-US": "SCI42069",
+                  th: "ว42069",
+                },
+                name: {
+                  "en-US": {
+                    name: "Additional Science",
+                  },
+                  th: { name: "วิทยาศาสตร์เพิ่มเติม" },
+                },
+                subjectSubgroup: {
+                  name: { "en-US": "Science", th: "วิทยาศาสตร์" },
+                  subjectGroup: {
+                    name: { "en-US": "Science", th: "วิทยาศาสตร์" },
+                  },
+                },
+              },
+            ],
+            contacts: [
+              {
+                id: 3,
+                name: {
+                  "en-US": "xx_ธนกร_อรรจนาวัฒน์_xx",
+                  th: "xx_ธนกร_อรรจนาวัฒน์_xx",
+                },
+                via: "facebook",
+                value: "",
+              },
+              {
+                id: 4,
+                name: {
+                  "en-US": "ธนกร_อรรจนาวัฒน์xd",
+                  th: "ธนกร_อรรจนาวัฒน์xd",
+                },
+                via: "line",
+                value: "",
+              },
+              {
+                id: 5,
+                name: {
+                  "en-US": "Thanacord",
+                  th: "Thanacord",
+                },
+                via: "discord",
+                value: "",
+              },
+            ],
+          },
+        },
+      ],
+    },
+  ];
+
+  const [showMain, setShowMain] = useState(teacherList[0].content[0].content);
+
+  showMain.contacts.sort(function (a, b) {
+    if (a.via < b.via) {
+      return -1;
+    }
+    if (a.via > b.via) {
+      return 1;
+    }
+
+    // names must be equal
+    return 0;
+  });
+
+  return (
+    <ListLayout
+      show={true}
```

Comment:
At the moment this page is completely unusable on mobile. The `show` attribute determines whether the Main Section is displayed on mobile or not. The convention is to control this attribute with a state called `showMain`, which would be set to `true` when a List Item is tapped, and to `false` when the back button on Title is tapped (`backGoesTo` attribute also accepts a function).

In List Item:
```jsx
<button
  onClick={() => {
    onClick();
    setShowMain(true);  // <--
    setMainContent(content);
  }}
  className="!w-full"
>
```
In Title:
```jsx
<Title
  name={{ title: t("title") }}
  pageIcon={<MaterialIcon icon="school" />}
  backGoesTo={showMain ? () => showMain(false) : "/account/login"}  // <--
  LinkElement={Link}
/>
```

Since you already have a state called `showMain` for the content of the Main Section, you should consider renaming it to `mainContent` or something like that.